src/sysext/sysext.c

```diff
@@ -478,11 +509,19 @@ static int merge_subprocess(Hashmap *images, const char *workspace) {
                 return r;
 
         /* Acquire host OS release info, so that we can compare it with the extension's data */
-        r = parse_os_release(
-                        arg_root,
-                        "ID", &host_os_release_id,
-                        "VERSION_ID", &host_os_release_version_id,
-                        "SYSEXT_LEVEL", &host_os_release_sysext_level);
+        if (!syscfg_setting)
+                r = parse_extension_os_release(
+                                arg_root,
+                                "ID", &host_os_release_id,
+                                "VERSION_ID", &host_os_release_version_id,
+                                "SYSEXT_LEVEL", &host_os_release_sysext_level);
+        else
+                r = parse_configuration_os_release(
+                                arg_root,
+                                "ID", &host_os_release_id,
+                                "VERSION_ID", &host_os_release_version_id,
+                                "SYSCFG_LEVEL", &host_os_release_syscfg_level);
```

Comment:
add helper:

```c
static inline const char* syscfg_or_sysext_level(void) {
        return syscfg_setting ? "SYSCFG_LEVEL" : SYSEXT_LEVEL";
}
```

and then pass as 6th argument here, so that you need a single invocation of parse_cofniguration_os_release()

Reply:
(and of course a herlp:

```
static typeof(parse_extension_os_release) parse_syscfg_or_sysctl_os_release(void) {
        return syscfg_setting ? parse_syscfg_os_release : parse_sysext_os_release;
}
```

So that you can call it:

```c
r = parse_syscfg_or_sysctl_os_release()(.);
```